templates/eks/amazon-eks-nodepool-cf.yaml

```diff
@@ -654,14 +670,52 @@ Resources:
+                - "\n"
+                -
+                    - "#!/bin/bash"
+                    - "set -o xtrace"
+                    - !Sub "if [ \"${UseInstanceStore}\" == \"true\" ]; then"
+                    - "  SSD_NVME_DEVICE_LIST=($(lsblk -l -o name,model | grep \"Instance Storage\" | cut -d \" \" -f 1 | sed -e \"s/.*/\\/dev\\/&/\" || true))"
+                    - "  SSD_NVME_DEVICE_COUNT=${#SSD_NVME_DEVICE_LIST[@]}"
+                    - "  RAID_DEVICE=${RAID_DEVICE:-/dev/md0}"
+                    - "  RAID_CHUNK_SIZE=${RAID_CHUNK_SIZE:-512}  # Kilo Bytes"
+                    - "  FILESYSTEM_BLOCK_SIZE=${FILESYSTEM_BLOCK_SIZE:-4096}  # Bytes"
+                    - "  STRIDE=$(expr $RAID_CHUNK_SIZE \\* 1024 / $FILESYSTEM_BLOCK_SIZE || true)"
+                    - "  STRIPE_WIDTH=$(expr $SSD_NVME_DEVICE_COUNT \\* $STRIDE || true)"
+                    - "  case $SSD_NVME_DEVICE_COUNT in"
+                    - "  \"0\")"
+                    - "     echo 'No devices found of type \"Amazon EC2 NVMe Instance Storage\"'"
+                    - "     ;;"
+                    - "  \"1\")"
+                    - "     echo \"Format nvme device $SSD_NVME_DEVICE_LIST\""
+                    - "     mkfs.ext4 -m 0 -b $FILESYSTEM_BLOCK_SIZE $SSD_NVME_DEVICE_LIST"
+                    - "     DEVICE=$SSD_NVME_DEVICE_LIST"
+                    - "     ;;"
+                    - "  *)"
+                    - "     yum install mdadm -y"
+                    - "     mdadm --create --verbose $RAID_DEVICE --level=0 -c ${RAID_CHUNK_SIZE} --raid-devices=${#SSD_NVME_DEVICE_LIST[@]} ${SSD_NVME_DEVICE_LIST[*]}"
+                    - "     while [ -n \"$(mdadm --detail $RAID_DEVICE | grep -ioE 'State :.*resyncing')\" ]; do"
+                    - "       echo \"Raid is resyncing..\""
+                    - "       sleep 1"
+                    - "     done"
+                    - "     echo \"Raid0 device $RAID_DEVICE has been created with disks ${SSD_NVME_DEVICE_LIST[*]}\""
+                    - "     mkfs.ext4 -m 0 -b $FILESYSTEM_BLOCK_SIZE -E stride=$STRIDE,stripe-width=$STRIPE_WIDTH $RAID_DEVICE"
+                    - "     DEVICE=$RAID_DEVICE"
+                    - "     ;;"
+                    - "  esac"
+                    - "  if [ $SSD_NVME_DEVICE_COUNT -gt 0 ]; then"
+                    - "    mkdir /media/local"
+                    - "    mount -o defaults,noatime,discard,nobarrier $DEVICE /media/local"
+                    - "    mkdir /media/local/kubelet"
+                    - !Sub "    /etc/eks/bootstrap.sh ${ClusterName} --kubelet-extra-args '--root-dir /media/local/kubelet ${KubeletExtraArguments}'"
+                    - "  else"
+                    - !Sub "  /etc/eks/bootstrap.sh ${ClusterName} --kubelet-extra-args '${KubeletExtraArguments}'"
+                    - "  fi"
+                    - "else"
+                    - !Sub "  /etc/eks/bootstrap.sh ${ClusterName} --kubelet-extra-args '${KubeletExtraArguments}'"
```

Comment:
shouldn't we add BootstrapArguments here (and the other branches)? even if we don't use it, the parameter is kept in the template, and it's not even marked as deprecated or unused in the description

Reply:
I didn't removed it for compatibility reasons and may be we will use it some time, I can append to the end of param list

Reply:
I'd do that. Or replace the description of the parameter with something like "deprecated, not used"